Bursts/matlab/getBinnedBurstSpikes.m

```diff
@@ -1,58 +1,62 @@
+% Last updated: 02/05/2022
+% Last updated by: Vu T. Tieu (vttieu1995@gmail.com)
 function getBinnedBurstSpikes(h5dir)
-% h5dir = '/CSSDIV/research/biocomputing/data/tR_1.9--fE_0.98'
+% h5dir = '/CSSDIV/research/biocomputing/data/tR_1.0--fE_0.90'              this line is used for testing
 if ~exist([h5dir '/Binned'], 'dir')
        mkdir([h5dir '/Binned'])
 end
 head = 10;                      % number of bins before burst start bin
 tail = 0;                       % number of bins after burst end bin
 binSize = 100;                  % one bin becomes one frame
-P = (hdf5read([h5dir '.h5'], 'spikesProbedNeurons'))';
-bbInfoFile = [h5dir '/binnedBurstInfo.csv'];
-b = csvread(bbInfoFile,1,1);
-n_bursts = length(b);                   % number of bursts identified
-n_neurons = length(P(1,:));             % number of neurons in simulation
-ended = length(P(:,1));
+spikesProbedNeurons = (h5read([h5dir '.h5'], '/spikesProbedNeurons'))';     % firing times & neuron IDs
+binnedBurstInfoFilePath = [h5dir '/allBinnedBurstInfo.csv'];
+binnedBurstsData = csvread(binnedBurstInfoFilePath,1,1);
+nBursts = length(binnedBurstsData);                         % number of bursts identified
+nNeurons = length(spikesProbedNeurons(1,:));                % number of neurons in simulation (this line gets the number of columns)
+ended = length(spikesProbedNeurons(:,1));                   % gets number of rows in spikesProbedNeurons
 % ------------------------------------------------------------------------
 % Get spike info from spikesProbedNeurons for each burst
 % ------------------------------------------------------------------------
-% every column in "frames" is a flatten image vector represents 100x100 
+% every column in "binnedBurstFrames" is a flatten image vector represents 100x100 
 % image (Matlab is column-major). To see what triggers a burst, save 10 
 % extra bins before burst starts 
 % (NOTE: bin size = 10ms, time step size = 0.1 ms)
 % ------------------------------------------------------------------------
-for i = 1:n_bursts
+for iBurst = 1:nBursts
     % get burst info from 10 bins before burst
-    frames = zeros(n_neurons, b(i,3)+head+tail, 'uint16');
-    s_time = (b(i,1)-head+1)*100;       % start bin# -> start time step
-    e_time = (b(i,2)+tail)*100;         % end bin# -> end time step  
-    % search spike time for each neuron (column in P)
-    for j = 1:n_neurons 
-        % first non-zero element in column i that is >= s_time
-        start = find(P(:,j) >= s_time, 1); 
+    binnedBurstFrames = zeros(nNeurons, binnedBurstsData(iBurst,3)+head+tail, 'uint16');
+    % converting start bin and end bin to include time step size
+    startingTimeStep = (binnedBurstsData(iBurst,1)-head+1)*100;         % start bin# -> start time step
+    endingTimeStep = (binnedBurstsData(iBurst,2)+tail)*100;             % end bin# -> end time step  
+    % search spike time for each neuron (column in spikesProbedNeurons)
+    for jNeuron = 1:nNeurons 
```

Comment:
I tested running this function with 1, 10, and 20 bursts, with 3 samples each. It averages to a little more than 17 seconds per burst. It would take around 42 hours with 9000 bursts.

Reply:
After we merge this branch, let's see if we can improve this function's efficiency.